packages/server/tests/acceptance/ws/subscribeNewHeads.spec.ts

```diff
@@ -150,4 +120,74 @@ describe('@web-socket Acceptance Tests', async function () {
       await new Promise((resolve) => setTimeout(resolve, 500));
     });
   });
+
+  describe('Subscriptions for newHeads', async function () {
+    it('should subscribe to newHeads and receive a valid JSON RPC response', (done) => {
+      process.env.WS_NEW_HEADS_ENABLED = 'true';
+      const webSocket = new WebSocket(WS_RELAY_URL);
+      const subscriptionId = 1;
+      webSocket.on('open', function open() {
+        webSocket.send(
+          JSON.stringify({
+            id: subscriptionId,
+            jsonrpc: '2.0',
+            method: 'eth_subscribe',
+            params: ['newHeads'],
+          }),
+        );
+      });
+
+      let responseCounter = 0;
+      webSocket.on('message', function incoming(data) {
+        const response = JSON.parse(data);
+        console.log(`responseCounter: ${responseCounter}, response: ${JSON.stringify(response)}`);
+        responseCounter++;
+        try {
+          // Validate the structure of the JSON RPC response
+          // first response is the subscription ID
+          if (responseCounter === 1) {
+            expect(response).to.have.property('jsonrpc', '2.0');
+            expect(response).to.have.property('id', 1);
+            expect(response).to.have.property('result'); // Subscription ID
+            // Ensure the result is a non-empty string (a valid subscription ID)
+            expect(response.result).to.be.a('string').that.is.not.empty;
+          } else {
+            expect(response).to.have.property('jsonrpc', '2.0');
+            expect(response).to.have.property('method', 'eth_subscription');
+            expect(response).to.have.property('params');
+            expect(response.params).to.have.property('result');
+            expect(response.params.result).to.have.property('difficulty');
+            expect(response.params.result).to.have.property('extraData');
+            expect(response.params.result).to.have.property('gasLimit');
+            expect(response.params.result).to.have.property('gasUsed');
+            expect(response.params.result).to.have.property('logsBloom');
+            expect(response.params.result).to.have.property('miner');
+            expect(response.params.result).to.have.property('nonce');
+            expect(response.params.result).to.have.property('number');
+            expect(response.params.result).to.have.property('parentHash');
+            expect(response.params.result).to.have.property('receiptsRoot');
+            expect(response.params.result).to.have.property('sha3Uncles');
+            expect(response.params.result).to.have.property('stateRoot');
+            expect(response.params.result).to.have.property('timestamp');
+            expect(response.params.result).to.have.property('transactionsRoot');
+            expect(response.params.result).to.have.property('hash');
+            expect(response.params).to.have.property('subscription');
+          }
+
+          if (responseCounter > 1) {
```

Comment:
To be clear this will mean the test will only ever confirm 2 responses.
The 1st to get the subscription ID and the 2nd to get the 1st block. You'll never confirm more than 1 blocks details.

You should definitely be testing for 0,1 and n numbers of blocks cases.

Reply:
Yes, it's a subscription so I need to limit the number of responses I test and focus on the content of those responses.